src/query.rs

```diff
@@ -55,52 +44,49 @@ impl<T: Eq + Hash + Copy + Debug> Index<T> {
                         &self.arena_index,
                     );
                     if let Some(term_node_index) = term_node_option {
-                        let document_frequency =
-                            self.disconnect_and_count_documents(term_node_index, removed);
+                        let document_frequency = self.count_documents(term_node_index);
                         let term_node = self.arena_index.get(term_node_index).unwrap();
                         if let Some(term_node_option_first_doc) = term_node.first_doc {
-                            if document_frequency > 0 {
```

Comment:
Why was this removed?  The reason for this if statement is a purely a performance once. It does not make sense to process the term any longer if it does not mach any documents

Reply:
My bad, that has been restored.